client/extensions/woocommerce/app/order/order-details.js

```diff
@@ -24,13 +27,57 @@ class OrderDetails extends Component {
 	}
 
 	state = {
-		showRefundDialog: false,
+		status: this.props.order.status || false,
 	}
 
-	toggleRefundDialog = () => {
-		this.setState( {
-			showRefundDialog: ! this.state.showRefundDialog,
-		} );
+	updateStatus = ( event ) => {
+		this.setState( { status: event.target.value } );
+		// Send the order back to the parent component
+		this.props.onUpdate( { status: event.target.value } );
+	}
+
+	renderStatus = () => {
+		const { order, translate } = this.props;
+		const classes = `order__status is-${ order.status }`;
+		const statuses = [ {
```

Comment:
Not necessary for v1/this PR but we should consider having a helper function for these labels, since we do something similar on the order list. Maybe add a `TODO`?